From 4162388dc8e8759b95e4ddcbad8a5d45a2104e4f Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Wed, 29 May 2024 15:33:36 +0200
Subject: [PATCH] find MPI

---
 cpp/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index e50fa61816..6ef41de071 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -84,6 +84,13 @@ add_feature_info(
   "Enable UFCx discovery using Python. Disable if UFCx should be found using a CMake config file."
 )
 
+# explicitly find MPI
+find_package(MPI)
+message(STATUS "MPI_C ${MPI_C_FOUND} ${MPI_C_COMPILER} ${MPI_C_LIBRARIES}")
+message(STATUS "MPI_CXX ${MPI_CXX_FOUND} ${MPI_CXX_LIBRARIES}")
+set(MPI_CXX_LIBRARIES "impi" CACHE STRING "" FORCE)
+message(STATUS "MPI_CXX ${MPI_CXX_COMPILER} ${MPI_CXX_FOUND} ${MPI_CXX_LIBRARIES}")
+get_target_property(OUT MPI:CXX LINK_LIBRARIES)
+message(STATUS "MPI:CXX ${OUT}")
 # ------------------------------------------------------------------------------
 # Enable or disable optional packages
 
-- 
2.42.0

