diff --git a/cpp/cmake/modules/FindSCOTCH.cmake b/cpp/cmake/modules/FindSCOTCH.cmake
index eea445f4bb..4aa72f3e2a 100644
--- a/cpp/cmake/modules/FindSCOTCH.cmake
+++ b/cpp/cmake/modules/FindSCOTCH.cmake
@@ -125,11 +125,11 @@ find_library(
   DOC "The PTSCOTCH-ERROR library"
 )

-set(SCOTCH_LIBRARIES ${PTSCOTCH_LIBRARY})
+set(SCOTCH_LIBRARIES ${PTSCOTCH_LIBRARY} ${SCOTCH_LIBRARY})
 if(PTESMUMPS_LIBRARY)
   set(SCOTCH_LIBRARIES ${SCOTCH_LIBRARIES} ${PTESMUMPS_LIBRARY})
 endif()
-set(SCOTCH_LIBRARIES ${SCOTCH_LIBRARIES} ${PTSCOTCHERR_LIBRARY})
+set(SCOTCH_LIBRARIES ${SCOTCH_LIBRARIES} ${PTSCOTCHERR_LIBRARY} ${SCOTCHERR_LIBRARY})

 # Basic check of SCOTCH_VERSION which does not require compilation
 if(SCOTCH_INCLUDE_DIRS)